robotframework-ls/src/robotframework_ls/impl/ast_utils.py

```diff
@@ -479,6 +488,25 @@ def create_keyword_usage_info(stack, node) -> Optional[KeywordUsageInfo]:
     return None
 
 
+def iter_variable_usage_tokens(ast) -> Iterator[VariableUsageInfo]:
+    """
+    Iterates through all the places where a variable name is being used, providing
+    the stack, node, token and name.
+    """
+    from robot.api import Token
+
+    for stack, node in _iter_nodes(ast, recursive=True):
+        if node.__class__.__name__ in ["KeywordCall", "Variable"]:
+            for token in node.tokens:
+                usage_info = None
+                if token.type == Token.ARGUMENT:
+                    variable_name = _RF_VARIABLE.findall(token.value)
+                    if variable_name and not _RF_NUM_CONST.findall(variable_name[0]):
+                        usage_info = VariableUsageInfo(tuple(stack), node, token, variable_name[0])
+                        if usage_info is not None:
```

Comment:
I think that this `if usage_info is not None` is not needed here (since the instance was just created).